### PR TITLE
Fixes for global promotion

### LINKED ISSUE #148: problems with SB_PLL40_2_PAD and promote_globals()
Observeable on https://github.com/swetland/cpu16/ as of commit ffc013add5208852276d815ab3af5dca26684764

foss toolchain(s) are ToT as of today:
```
icestorm 9671b760f84ca4006f0ef101a3e3b201df4eabb5
nextpnr ff978570b162844de61dfa4ea9e30db6904b0b1f
arachnepnr 840bdfdeb38809f9f6af4d89dd7b22959b176fdd
yosys ab97eddee9b0ea0a772660731fe4c3270d2564e5
```

build w/ `make USE_NEXTPNR=true`, output is in `out/ice40.asc` `out/ice40.bin` etc

If the full design is built, the 12m clock from the PLL's GLOBALA (passthru) is visible on out1 (pin) but GLOBALB on out2 is flatline.  Exposing the lock signal showed it high in both cases.  Building the design with `hdl/ice40.v`'s `define WITH_CPU commented out (leaving only the SPI debug interface and VGA blck) results in a functional sub-design.

Commenting out `promote_globals()` in `ice40/pack.cc` per tnt's suggestion resolves the issue (but only uses 2 globals -- for the two PLL outputs).

My guess is `promote_globals()` is managing to clobber the GLOBALB PLL output...

### PR DESCRIPTION
**Pull request by smunaut**
Fixes #148 

@swetland If you could confirm that :)

### CODE REVIEW

File: common/placer1.cc [hunk 1]
```diff
@@ -350,6 +351,8 @@ class SAPlacer
                     log_error("failed to place cell '%s' of type '%s'\n", cell->name.c_str(ctx), cell->type.c_str(ctx));
                 --iters;
                 ctx->unbindBel(ripup_target->bel);
+                if (ripup_target->type == id_SB_GB)
```
Comment (daveshah1) [inline common/placer1.cc]: This will break the ECP5 build, where SB_GB doesn't exist (this seems just to be debugging anyway)

**Comment by daveshah1** [comment]
LGTM bar one change

File: common/placer1.cc (hunk 1 above)
Comment (smunaut) [reply common/placer1.cc]: Doh .. yeah, not sure how I missed that :/

**Comment by smunaut** [comment]
Just forced pushed an updated branch without that debug print.

**Comment by swetland** [comment]
Confirmed.  This PR resolves #148 for me.  Thanks!

**Comment by daveshah1** [comment]
Thanks again for your work @smunaut!

**Pull request merged**